### PR TITLE
Hide Hive system schemas in Hudi

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Hide Hive system schemas in Hudi.

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Hudi
* Hide Hive system schemas. ({issue}`14510`)
```

### CODE REVIEW

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+public class TestHudiOnHive3
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String bucketName = "test-hudi-connector-" + randomTableSuffix();
+        HiveMinioDataLake hiveMinioDataLake = closeAfterClass(new HiveMinioDataLake(bucketName, HIVE3_IMAGE));
```
Comment (findepi) [inline plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java:38]: I'd change one or all subclasses of BaseHudiMinioConnectorTest

i don't know why we would want to test with hdp2.6 more than hdp3.1
(i know hdp2.6 is the "default image", but maybe that's a mistake)

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        hiveMinioDataLake.start();
+        hiveMinioDataLake.getMinioClient().ensureBucketExists(bucketName);
```
Comment (findepi) [inline plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java:40]: `HiveMinioDataLake.start` does that, doesn't it?
https://github.com/trinodb/trino/blob/e73567b47412105206c9272e8edc95f98c771a4f/plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/HiveMinioDataLake.java#L172

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                HUDI_MINIO_TESTS.getSchemaName(),
+                ImmutableMap.of(),
+                new TpchHudiTablesInitializer(COPY_ON_WRITE, ImmutableList.of()),
+                hiveMinioDataLake);
+    }
+
+    @Test
+    public void testHideHiveSysSchema()
+    {
+        assertThat(computeActual("SHOW SCHEMAS").getOnlyColumnAsSet()).doesNotContain("sys");
```
Comment (findepi) [inline plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java:53]: it's quite expensive test or this assertion alone.

i'd probably abstain from adding a test for this, unless we can hijack some existing one, and not pay setup cost.

**Review by findepi — APPROVED** [review]

**Review by findinpath — APPROVED** [review]

File: plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiOnHive3.java]: Added a preparatory commit to use hdp3.1 in the all subclasses.

**Review by codope — APPROVED** [review]

**Pull request merged**